cEP-0020.md

```diff
@@ -0,0 +1,529 @@
+  - Name: Beginner
+  - Points_required = 20
+  - Privileges:
+    - Get assigned to a difficulty low issue
+- Level-IV:
+  - Name: Intermediate
+  - Points_required = 50
+  - Privileges:
+    - Mark your own PR as wip
+- Level-V:
+  - Name: Cool
+  - Points_required = 100
+  - Privileges:
+    - Mark a PR by another newcomer as wip
+    - Get assigned to difficulty medium issue
+- Level-VII:
+  - Name: Expert
+  - Points_required = 200
+  - Privileges
+    - Assign newcomers to difficulty newcomer issue
+    - Assign newcomers to difficulty low issue
+- Level-VIII:
+  - Name: Legend
+  - Points_required = 350
+  - Privileges
+    - Invite newcomers to org
+    - Assign newcomers to difficulty medium issue
+    - Permission to `ack` and `unack` on the pull requests
+
+Implementation of privileges is beyond scope of the
+gamification system. The gamification system will merely
+expose a list of privileges that each person has, and other
+systems are expected to adapt to use these privileges.
+E.g.: [corobo](https://github.com/coala/corobo)
+
+While implementing privileges, a first manual check should
+be done for testing purpose, and then automated only if there
+is no serious risk or there is a good risk reaction mechanism
+in place.
+
+### Badges
+
+Another way of explicitly nudging a newcomer to action is to
+award badges for completing tasks. Badges enable newcomers to
+follow their performance and compare to other newcomers.
+Badges will be awarded after performing certain activities by
+the newcomers:
+
+- The bug finder:
+  - Details: The one who find bugs in the existing codebase
+  - Activities:
+    - Create at least one issue which labelled as bug
+- The bear hunter:
+  - Details: The one who create issues about new bears
+  - Activities:
+    - Create at least one bear proposal issue
+- The bear writer:
+  - Details: The one who write new bears
+  - Activities:
+    - Merged at least 2 bear proposal PRs
+- The helper:
+  - Details: The one who answers every newcomer questions on Gitter
+  - Activities:
+    - At least 20 answers given on coala Gitter rooms
+- The coala all-rounder:
+  - Details: Has met the prerequisites to apply for the coala developer role
+  - Activities:
+    - Introduce `.coafile` in user own repository
+    - Merge a difficulty/newcomer Pull Request
+    - Review at least a difficulty/newcomer Pull Request
+    - Merge a difficulty/low Pull Request
+    - Review at least a difficulty/low or higher Pull Request
+
+So, to be eligible to be a developer at coala, a newcomer must
+complete all the activities mentioned in the `coala all-rounder`
+badge. But in case if some of the newcomers are not able to be
+eligible to apply for coala developer role, they could take away
+the other badges.
+
+Names and metrics of tracked activities will also be shown
+on the leaderboard along with the number of times they
+have been done. That will help maintainers to varify if an
+automatically awarded badge is correct or not.
```

Comment:
How about points for badges?

Reply:
A newcomer can earn points by performing any kind of activity. 

And the badges are given here to show a newcomer's skill in a particular type of activity.

If we give badges based on points then the activities will not have much more meaning.

Reply:
We're keeping badges distinct from the points/levels , as we will be revising our points/levels after go-live , and that will mean people's points may change, and they may even drop down a level , but a badge shouldnt be revoked once it has been given.
If a badge "should" be changed, that would need to be done by versioning the badge, and maybe even renaming the old badge to show it wasnt well designed.

@sks444 I dont think we have anything about this distinction yet.  Probably belongs up much higher as a guiding principle.